### PR TITLE
Fix lsa to madx writer

### PR DESCRIPTION
**Pull request by JoschD**
- LSA to MADX sign convention
 - Better trim naming
 - check madx names for allowed characters
 - option to init all variables
 - machine settings info takes ISO time

### CODE REVIEW

File: pylhc/lsa_to_madx.py [hunk 1]
```diff
@@ -135,7 +140,38 @@ def parse_knobs_and_trim_values_from_file(knobs_file: Path) -> Dict[str, float]:
...
+
+    """
+    # Test for skewness:
+    match = re.match(r"^.c?[qsodt]s", madx_name)  # k, then maybe a "c", then type, then skew
+    if match is not None:
+        LOG.debug(f"{madx_name} belongs to a skew magnet: Sign is inverted betwen LSA and MAD-X")
+        return -1
+
+    # Test for Q2
+    if madx_name.startswith("ktqx2"):
```
Comment (fsoubelet) [inline pylhc/lsa_to_madx.py:161]: `.lower()` maybe?

File: pylhc/lsa_to_madx.py [hunk 2]
```diff
@@ -146,6 +182,12 @@ def get_madx_script_from_definition_dataframe(deltas_df: tfs.TfsDataFrame, lsa_k
             can be obtained with `~pylhc.data_extract.lsa.LSAClient.get_knob_circuits`.
         lsa_knob (str): The complete ``LSA`` name of the knob, including any ``LHCBEAM[12]?/`` part.
         trim (float): The trim value to write for the knob. Defaults to 1.
+        by_reference (bool): If true, creates an _init variable and uses the `:=` operator
```
Comment (fsoubelet) [inline pylhc/lsa_to_madx.py:185]: maybe using the words "deferred expression" can help the user

File: tests/unit/test_lsa_to_madx.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
```
Comment (fsoubelet) [inline tests/unit/test_lsa_to_madx.py:2]: maybe we can also do a version of the test without the deferred expressions

Comment (JoschD) [reply tests/unit/test_lsa_to_madx.py:2]: It doesn't test much anyway. And I don't like these tests, as you need to change the file it compares to as soon as you do any change in the code. A proper test would be running the resulting file in madx/cpymad and checking that the variables change as expected, but I don't have the time to implement this.

**Review by fsoubelet — APPROVED** [review]

**Pull request merged**